virttest/utils_misc.py

```diff
@@ -4022,26 +4023,36 @@ def get_model_features(model_name):
     """
     features = []
     conf = "/usr/share/libvirt/cpu_map.xml"
-
+    conf_dir = "/usr/share/libvirt/cpu_map/"
     try:
-        output = open(conf, 'r').read()
-        root = ET.fromstring(output)
-        # Find model
-        for model_n in root.findall('arch/model'):
-            if model_n.get('name') == model_name:
-                model_node = model_n
-                for feature in model_n.findall('feature'):
-                    features.append(feature.get('name'))
-                break
-        # Handle nested model
-        nested_model = model_node.find('model')
-        if nested_model is not None:
-            nested_model_name = nested_model.get('name')
+        if os.path.isfile(conf) and os.path.exists(conf):
+            output = open(conf, 'r').read()
```

Comment:
better to use `with` statements.

Reply:
thanks for helping me review the code.
I will fix the code soon.

Reply:
thanks for helping me review the code.
I will fix the code soon.